### PR TITLE
"Finished" check now considers the set IO mode

### PR DESCRIPTION
**Pull request by ineternet**
Hopefully fix for #1705, didn't have the time to test this, but it should work considering the issue description. Since the bug was my fault, I assumed it's my function/check that was wrong.

### CODE REVIEW

**Comment by raoulvdberge** [comment]
Can you test this please?

**Comment by ineternet** [comment]
It seems like there's more than just what I thought it is behind it (extract mode doesn't work *at all* now), I'm checking this out right now and testing it.

**Comment by ineternet** [comment]
This should be working and up to date with the current manipulator now. Tested, both modes work again.

File: src/main/java/com/raoulvdberge/refinedstorage/apiimpl/network/node/diskmanipulator/NetworkNodeDiskManipulator.java [hunk 1]
```diff
@@ -195,19 +195,23 @@ private void insertItemIntoNetwork(IStorageDisk<ItemStack> storage, int slot) {
...
             return true;
         }
 
+        //In Extract mode, we just need to check if the disk is full or not.
+        if (ioMode == IO_MODE_EXTRACT)
+            if (storage.getStored() == storage.getCapacity()) {
+                moveDriveToOutput(slot)
+                return true;
+            }
+            else return false;
```
Comment (raoulvdberge) [inline src/main/java/com/raoulvdberge/refinedstorage/apiimpl/network/node/diskmanipulator/NetworkNodeDiskManipulator.java]: Please fix this formatting

File: src/main/java/com/raoulvdberge/refinedstorage/apiimpl/network/node/diskmanipulator/NetworkNodeDiskManipulator.java [hunk 2]
```diff
@@ -298,11 +297,19 @@ private void insertFluidIntoNetwork(IStorageDisk<FluidStack> storage, int slot)
...
```
Comment (raoulvdberge) [inline src/main/java/com/raoulvdberge/refinedstorage/apiimpl/network/node/diskmanipulator/NetworkNodeDiskManipulator.java]: Here too

File: src/main/java/com/raoulvdberge/refinedstorage/apiimpl/network/node/diskmanipulator/NetworkNodeDiskManipulator.java [hunk 3]
```diff
@@ -195,19 +195,23 @@ private void insertItemIntoNetwork(IStorageDisk<ItemStack> storage, int slot) {
...
-        if (storage.getStored() == 0) {
+        if (ioMode == IO_MODE_INSERT && storage.getStored() == 0) {
             moveDriveToOutput(slot);
             return true;
         }
 
+        //In Extract mode, we just need to check if the disk is full or not.
+        if (ioMode == IO_MODE_EXTRACT)
+            if (storage.getStored() == storage.getCapacity()) {
+                moveDriveToOutput(slot)
```
Comment (raoulvdberge) [inline src/main/java/com/raoulvdberge/refinedstorage/apiimpl/network/node/diskmanipulator/NetworkNodeDiskManipulator.java]: I don't believe you tested this. This line doesn't even compile, that's also why the build is failing. How did you "test" without this line being able to compile?

**Comment by ineternet** [comment]
I tested this before the 3rd commit, not after, and assumed the code should work (which, after testing it now in the current state, it does), just forgot semicolons there.

**Comment by raoulvdberge** [comment]
The formatting is still messed up. Please use braces.

**Comment by ineternet** [comment]
its not messed up, i just like it that way :/
changed it anyways

**Comment by raoulvdberge** [comment]
It's STILL messed up:
x

You're contributing to a open source project, you need to follow the coding style of said project, whether you like it or not ...

**Comment by ineternet** [comment]
done that now

**Comment by raoulvdberge** [comment]
Thank you.

**Pull request merged**